sycl/source/detail/devicelib/CMakeLists.txt

```
@@ -0,0 +1,68 @@
+# Place device libraries near the libsycl.so library in a build
+# directory
+if (MSVC)
+    set(binary_dir "${CMAKE_RUNTIME_OUTPUT_DIRECTORY}")
+else()
+    set(binary_dir "${CMAKE_LIBRARY_OUTPUT_DIRECTORY}")
+endif()
+
+set(clang $<TARGET_FILE:clang>)
+
+set(compile_opts
+    # suppress an error about SYCL_EXTERNAL
+    -Wno-error=sycl-strict -Wno-sycl-strict
+    # for CL/__spirv/spirv_vars.hpp
+    -I${sycl_inc_dir})
+
+if (MSVC)
+    set(devicelib-obj-file ${binary_dir}/libsycl-msvc.o)
+    add_custom_command(OUTPUT ${devicelib-obj-file}
+                       COMMAND ${clang} -fsycl -c
+                               ${compile_opts}
+                               ${CMAKE_CURRENT_SOURCE_DIR}/msvc_wrapper.cpp
+                               -o ${devicelib-obj-file}
+                       MAIN_DEPENDENCY msvc_wrapper.cpp
+                       DEPENDS wrapper.h clang
+                       VERBATIM)
+else()
+    set(devicelib-obj-file ${binary_dir}/libsycl-glibc.o)
+    add_custom_command(OUTPUT ${devicelib-obj-file}
+                       COMMAND ${clang} -fsycl -c
+                               ${compile_opts}
+                               ${CMAKE_CURRENT_SOURCE_DIR}/glibc_wrapper.cpp
+                               -o ${devicelib-obj-file}
+                       MAIN_DEPENDENCY glibc_wrapper.cpp
+                       DEPENDS wrapper.h clang
+                       VERBATIM)
+endif()
+
+add_custom_command(OUTPUT ${binary_dir}/libsycl-fallback-cassert.spv
+                   COMMAND ${clang} -fsycl-device-only -S -Xclang -emit-llvm-bc
```

```suggestion
                   COMMAND ${clang} -fsycl-device-only -S
```

This should be enough

Moreover, you can just use `-fno-sycl-use-bitcode` to get SPIR-V file directly, see [docs](https://github.com/intel/llvm/blob/sycl/sycl/doc/SYCLCompilerUserManual.md#sycl-device-code-compilation)

---

Thanks, done.